data/src/main/java/com/linkedin/data/template/DataTemplateUtil.java

```diff
@@ -832,14 +903,46 @@ public static boolean hasCoercer(Class<?> klass)
       if (fromClass.isEnum())
         return object.toString();
     }
-    @SuppressWarnings("unchecked") DirectCoercer<T> coercer = (DirectCoercer<T>) _classToCoercerMap.get(fromClass);
+
+    return coerceCustomInput(object, fromClass);
+  }
+
+  public static Object coerceIntInput(Integer value)
```

Comment:
For all of these new methods you're introducing, consider the access level carefully. Do all of these need to be public? Can we reduce any of them to private, or package-private? If so, then we definitely should.

Reply:
All these util methods are invoked from generated classes and hence need to be public.